x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/Methods.java

```diff
@@ -47,7 +50,7 @@
  */
 public class Methods {
     static ExecutableElement findRequiredMethod(TypeElement declarationType, String[] names, Predicate<ExecutableElement> filter) {
-        ExecutableElement result = findMethod(declarationType, names, filter);
+        ExecutableElement result = findMethod(names, filter, declarationType, superClassOf(declarationType));
```

Comment:
Added support for inheritence in finding the static methods used in aggregations, because, for example, the SpatialCentroid can be configured in four different ways (geo vs cartesian, and doc-values vs source values). Used only single-level inheritance for simplicity.